### PR TITLE
Checkout: only use alt email in gsuite contact form

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* When checking out with gsuite in the cart, but _not_ any domain products, (1) only show the alternate email field in the summary contact details and (2) allow the user to update this field.

#### Testing instructions

Note that GSuite can only be tested in the live environment, with a domain that was purchased in the live environment - so you'll need to turn off store sandbox and use a real domain that does not already have gsuite added to test this.

* Enter checkout with a new gsuite subscription in your cart, but no domain products. You can do this on an account with an existing domain through `Manage > Domains` or directly with `/email/$siteurl/manage/$siteurl`.
* Verify that your "alternate email" is displayed in summary view, if it has already been saved for your user. If not (e.g. your test account doesn't already have gsuite on another domain), edit the contact details and add it.
* Try to set your alternate email to the email address being purchased and verify that you see an error message.

Before:

image

After (edit view):

image

After (summary view):

image

Error:

image

### CODE REVIEW

**Comment by stephanethomas** [comment]
Thanks for having a look to this issue! I've tested this pull request but I'm getting two email fields when trying to purchase G Suite for a domain already registered - when my domain contact information has an email pointing to that domain:

screenshot

As a side effect, clicking the `Continue` button without filling the form leads to the following error (with a wrong error message displayed for the wrong input field):

screenshot

I've also noticed a regression when the email address is not pointing to the domain registered. Previously, we would display that address in the summary:

screenshot

Now the email address is no longer displayed in the summary:

screenshot

When I click the `Edit` button, an `Email` field is displayed but unfortunately it is both empty and misaligned:

screenshot

Just to clarify, my original proposal was to simplify our code by removing exceptions around that alternate email (i.e. displaying a single input field all the time even if the email address we have on file doesn't point to the domain registered). I'm seeing the opposite with this pull request where we add more code and conditions. I was trying to make your life easier 😊.

I've got a crazy idea that I'd like to submit you: what if we separate the `G Suite account information` section from the `Contact information` section? We would display both sections when the user is purchasing a domain with G Suite (and just the `G Suite account information` section when the user is purchasing G Suite for an existing domain):

screenshot

Instead of having some G Suite code intermingled with some domain contact information code, we would have two separate components with very clear responsibilities. This should also make it easier for us to add a new component for Titan since it may come with its own rules. The only downside is that the checkout page would be longer. On the other hand, domain contact information and G Suite customer information are two distinct sets of information so it makes sense to have separate forms. What do you think?

**Comment by nbloomf** [comment]
Thank you for the thorough review @stephanethomas! This exposes some test cases I did not think to check; I'm adding them to my test documentation so I won't miss them in the future.

Regarding your suggestion about putting the GSuite fields in a separate step - we probably can't put them in a separate _step_ (this would add an extra required click, which we need to avoid), but we can put them in a separate component. There is even some precedent for this with the extra fields required for ccTLD purchases. I'll try that.

**Comment by sirbrillig** [comment]
Note: this fixes https://github.com/Automattic/wp-calypso/issues/45336

**Comment by nbloomf** [comment]
@stephanethomas - Payton and I paired to fix this issue. Do you mind doing another round of tests to make sure it behaves as expected? GSuite has enough dark corners that I'm not confident in my ability to catch all the problems.

File: client/components/domains/contact-details-form-fields/managed-contact-details-form-fields.jsx [hunk 1]
```diff
@@ -323,18 +323,7 @@ export class ManagedContactDetailsFormFields extends Component {
...
-		// those errors somewhere. However, only show them if the `alternateEmail`
-		// has not been entered.
-		if (
-			! customErrorMessage &&
-			this.props.contactDetailsErrors?.email &&
-			! this.props.contactDetails?.alternateEmail?.length > 0
-		) {
-			customErrorMessage = this.props.contactDetailsErrors.email;
-		}
+		const customErrorMessage = this.props.contactDetailsErrors?.alternateEmail;
```
Comment (stephanethomas) [inline client/components/domains/contact-details-form-fields/managed-contact-details-form-fields.jsx:326]: Not a big deal but this variable is no longer necessary - we could possibly inline that code.

File: client/my-sites/checkout/composite-checkout/components/wp-contact-form.js [hunk 2]
```diff
@@ -269,9 +268,7 @@ function DomainContactDetails( {
 		! hasDomainRegistration( responseCart ) &&
 		! hasTransferProduct( responseCart );
 	const getIsFieldDisabled = () => isDisabled;
-	const needsAlternateEmailForGSuite =
-		needsOnlyGoogleAppsDetails &&
-		needsExplicitAlternateEmailForGSuite( responseCart, contactDetails );
+	const needsAlternateEmailForGSuite = needsOnlyGoogleAppsDetails;
```
Comment (stephanethomas) [inline client/my-sites/checkout/composite-checkout/components/wp-contact-form.js:271]: Same remark, we could remove that variable and pass `needsOnlyGoogleAppsDetails` directly.

**Review by stephanethomas — COMMENTED** [review]
Sure thing! Here is a list of use cases that I've tested:

1. Purchase only a domain 
2. Purchase both a domain and G Suite
3. Purchase G Suite for an existing domain
4. Purchase G Suite for an existing domain where alternate email points to this domain
5. Purchase extra license(s) for an existing G Suite account

And a list of things that I've checked:

- [x] Name of the section
- [x] Text of the section
- [x] Error message when the email points to the domain being purchased
- [x] Error message when the alternate email points to the domain of G Suite account being purchased

I've noticed that we are still displaying a wrong text for use case #3 though:

> Registering a domain name requires valid contact information. Privacy Protection is included for all eligible domains to protect your personal information.

I've also realized that we ask the user for billing information when a user purchases extra licenses (use case #5):

screenshot

This is unrelated to those changes as it already happens in production. However, we probably should fix that given that we only ever need customer information when creating a brand new G Suite account, not when adding extra licenses. By the way, for some reason we don't display the actual mailboxes in the `Your order` section for extra licenses whereas we show them for new G Suite accounts.

**Comment by sirbrillig** [comment]
> I've noticed that we are still displaying a wrong text for use case #3 though

For reference, https://github.com/Automattic/wp-calypso/issues/45301 tracks that issue. Not that it can't be fixed here if we like.

**Comment by sirbrillig** [comment]
> I've also realized that we ask the user for billing information when a user purchases extra licenses

I'm glad you mentioned that! That's a requirement we don't have in the new checkout code. Added an issue specifically for that here: https://github.com/Automattic/wp-calypso/issues/45786

**Comment by sirbrillig** [comment]
>  By the way, for some reason we don't display the actual mailboxes in the Your order section for extra licenses whereas we show them for new G Suite accounts.

I just created https://github.com/Automattic/wp-calypso/issues/45787 to track that. Thank you!

**Review by sirbrillig — APPROVED** [review]
I think that given the discussion (and additional issues added above), this works as described. We can take care of the remaining issues in later PRs.

**Comment by stephanethomas** [comment]
>  We can take care of the remaining issues in later PRs.

Works for me 👍.

**Pull request merged**